### PR TITLE
xclicker: init at 1.5.0

### LINKED ISSUE #269012: Package request: XClicker
**Project description**

XClicker is a GUI application, that lets you set a specific position on your screen, and a custom time interval, in which the app sends mouse clicks onto this position. 

There seems to be no alternative, unless one proprietary solution, called Max Auto Clicker.  

**Metadata**

* homepage URL: https://xclicker.xyz/
* source URL: https://github.com/robiot/xclicker
* license: gpl3
* platforms: linux

**Priorities**

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/269012

This PR adds 1 new package `xclicker`

While packaging I found an malloc related issue, opened a PR for it in upstream (https://github.com/robiot/xclicker/pull/91)
Once it gets accepted, I'll use `fetchpatch`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mrcjkb** [comment]
Result of `nixpkgs-review pr 269031` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xclicker</li>
  </ul>
</details>

**Review by mrcjkb — APPROVED** [review]
##### Reviewed points

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] all depending packages build
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed

##### Comments

I do get the following crash when clicking the `settings` button:

```
❯ ./result/bin/xclicker

(xclicker:318346): Gtk-WARNING **: 23:17:06.251: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
*** buffer overflow detected ***: terminated
zsh: IOT instruction (core dumped)  ./result/bin/xclicker
```

Apart from that, it seems to work well.

File: pkgs/by-name/xc/xclicker/package.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  src = fetchFromGitHub {
+    owner = "robiot";
+    repo = "xclicker";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-3D49iMzCCT9Z2Pf5INHYFZusG0BQI7La7lLaSVM/4mc=";
+  };
+
+  # Change to fetchpatch after it's been upstreamed
+  patches = [ ./fix-malloc-size.patch ];
```
Comment (mrcjkb) [inline pkgs/by-name/xc/xclicker/package.nix:23]: suggestion: A URL to the upstream PR might be useful in the comment.

**Comment by TomaSajt** [comment]
> I do get the following crash when clicking the `settings` button:
> ...
Yeah, I didn't check the settings button.
It turns out I missed `wrapGAppsHook`, that made it work...
or it should have, but I found even more malloc size mistakes, so I updated the PR and the patch...

**Comment by TomaSajt** [comment]
The patch has been merged, changed to `fetchpatch`

**Comment by PassiveLemon** [comment]
Came across this PR when having the same buffer overflow problem in my own package.
Tested this and it works perfectly.

**Review by gepbird — APPROVED** [review]
Code and binary LGTM!

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**